crates/ra_ide/src/completion/complete_keyword.rs

```diff
@@ -82,6 +104,36 @@ pub(super) fn complete_expr_keyword(acc: &mut Completions, ctx: &CompletionConte
     acc.add_all(complete_return(ctx, &fn_def, ctx.can_be_stmt));
 }
 
+fn previous_non_triva_element(token: &SyntaxToken) -> Option<SyntaxElement> {
+    // trying to get first non triva sibling if we have one
+    let token_sibling = non_trivia_sibling(NodeOrToken::Token(token.to_owned()), Direction::Prev);
+    let mut wrapped = if let Some(sibling) = token_sibling {
+        sibling
+    } else {
+        // if not trying to find first ancestor which has such a sibling
+        let node = token.parent();
+        let range = node.text_range();
+        let top_node = node.ancestors().take_while(|it| it.text_range() == range).last()?;
+        let prev_sibling_node = top_node.ancestors().find(|it| {
+            non_trivia_sibling(NodeOrToken::Node(it.to_owned()), Direction::Prev).is_some()
+        })?;
+        non_trivia_sibling(NodeOrToken::Node(prev_sibling_node), Direction::Prev)?
+    };
+    // traversing the tree down to get the last token or node, i.e. the closest one
+    loop {
```

Comment:
I think you can avoid this `loop` if you use `SyntaxToken::prev_token` -- unlike `prev_sibling_or_token`, it works across parents.